### ISSUE TITLE
vim9script: error for some expressions involving parentheses after `execute`

### ISSUE THREAD

**Issue by bfrg** [vim/vim#6531]
**Describe the bug**
Expressions after `execute` are not always parsed correctly in `vim9script`.

**To Reproduce**
1. Run `vim --clean -S test.vim` with
   ```vim
   vim9script
   def Test(n: number)
       execute 'echomsg' (n ? '"yes"' : '"no"')
   enddef
   Test(1)
   ```
2. Result:
   ```
   Error detected while processing function <80><fd>R137_Test:
   line    1:
   E1085: Not a callable type: 'echomsg' (n ? '"yes"' : '"no"')
   ```
---

The above works fine in legacy functions:
```vim
function Test2(n)
    execute 'echomsg' (a:n ? '"yes"' : '"no"')
endfunction
call Test2(1)
```

The following two alternatives work though:
1. Concatenate the strings with a `..`:
   ```vim
   def Test(n: number)
       execute 'echomsg ' .. (n ? '"yes"' : '"no"')
   enddef
   ```
2. Remove the parentheses:
   ```vim
   def Test(n: number)
       execute 'echomsg' n ? '"yes"' : '"no"'
   enddef
   ```

**Expected behavior**
No error messages. Same behavior as in legacy Vim script.

**Environment**
Vim version `8.2.1287`

**Comment by lacygoill** [comment]
Fixed by [8.2.1293](https://github.com/vim/vim/releases/tag/v8.2.1293).

**Comment by lacygoill** [comment]
> Fixed by 8.2.1293.

Btw, the commit message displayed on github is wrong; it's a duplicate of the next commit 8.2.1294.  The correct commit message can be found at `:h version8`:

> Patch 8.2.1293
> Problem:    Vim9: :execute mixes up () expression and function call.
> Solution:   Do not skip white space when looking for the "(". (closes #6531)
> Files:      src/vim9compile.c, src/testdir/test_vim9_script.vim
